### ISSUE TITLE
Can not use `steps[currentStep].content` when `content` is `JSX`

### ISSUE THREAD

**Issue by hali241997** [elrumordelaluz/reactour#470]
**Describe the bug**
I've added some `JSX` inside the `content` of a step. Now I want to render it inside the `ContentComponent` using `steps[currentStep].content`. I've tried everything but it doesn't show up. But as long as I comment out the `ContentComponent`, it does show up. I need to show it inside the `ContentComponent`.

**To Reproduce**
1. Go to this sandbox link: https://codesandbox.io/s/keen-kowalevski-zopbhd
2. Click on the `Start Tour` button to start the tour. It will show the `JSX` that is added inside the `content` of the step.
3. Scroll down to 'ContentComponent' and uncomment it.
4. It will no longer show the `JSX` inside the `content` of the step.

**Expected behavior**
`JSX` to show up inside the `ContentComponent` by `steps[currentStep].content`.

**Screenshots**
https://user-images.githubusercontent.com/33332648/165378482-51cf70b7-a8ba-46ae-a682-cab19b000910.mov

**Desktop (please complete the following information):**
 - OS: MacOS
 - Browser: Chrome
 - Version: 12.3.1

**Additional context:**
1. I think I'm using `ContentComponent` the wrong way, please guide

**Comment by elrumordelaluz** [comment]
Hi @hali241997, thanks for open the _Issue_.

The reason is that when using `ContentComponent` you are responsible to check if the `step.content` prop is a `function` or not. 

You can do something like this:

```js

function ContentComponent(props) {
  const content = props.steps[props.currentStep].content;
  return typeof content === "function"
    ? content({ ...props, someOtherStuff: "Custom text" })
    : content;
}
```

[Here](https://codesandbox.io/s/pensive-moon-geyxus?file=/src/App.js) is a working example.

Will update the example in Readme to be clearer. Thanks to pointing this out!

**Comment by hali241997** [comment]
@elrumordelaluz This might work in `js` but it gives typing errors in `ts`. A better approach:
```
ContentComponent={({
                steps,
                currentStep,
                setCurrentStep,
                setIsOpen,
              }) => {
                const content = steps[currentStep].content;

                if (typeof content === "function") {
                  return (
                    <>
                      {content({
                        currentStep,
                        setCurrentStep,
                        setIsOpen,
                        transition: false,
                      })}
                    </>
                  );
                }
                return null;
              }}
```

Since `ContentComponent` expect `JSX`, so we wrap the `content` with `<></>`. The `content` only needs the following:
- `currentStep`
- `setCurrentStep`
- `setIsOpen`
- `transition`

I have set `transition` to `false` because it is passed as an optional prop from `ContentComponent`. For the rest, I passed as the same.
Updated codesandbox: https://codesandbox.io/s/keen-kowalevski-zopbhd

**Comment by elrumordelaluz** [comment]
Thank you @hali241997, mind giving an eye into the codebase in order to improve typing directly? 

Thanks again!

**Comment by hali241997** [comment]
@elrumordelaluz Sure I'll do that. Just a follow-up question. Is there any way where I can pass in custom props to `steps` props and use them accordingly inside `ContentComponent`?

**Comment by elrumordelaluz** [comment]
Something like in my previous example, but the right way using TS you mean?

**Comment by hali241997** [comment]
Let me rephrase. The `StepType` is this:
```
type StepType = {
    selector: string | Element;
    content: ReactElement | string | ((props: ContentProps) => void);
    position?: Position;
    highlightedSelectors?: string[];
    mutationObservables?: string[];
    resizeObservables?: string[];
    navDotAriaLabel?: string;
    stepInteraction?: boolean;
    action?: (elem: Element | null) => void;
    disableActions?: boolean;
    padding?: Padding;
    bypassElem?: boolean;
    styles?: StylesObj & PopoverStylesObj & MaskStylesObj;
}
```

What if someone wants to extend it? And then use that extension inside `CustomComponent`?
Like really passing the complete control to the user. Would that be possible?

**Comment by elrumordelaluz** [comment]
First should be extendible the`StepType` right? because `ContentComponent?: ComponentType<PopoverContentProps>` has inside `steps: StepType[]`.

<img width="451" alt="Screenshot 2022-04-28 at 09 34 08" src="https://user-images.githubusercontent.com/784056/165701123-34644882-ba3e-48e3-b618-580317094225.png">

Let me know if I am understanding well the point please.

**Comment by elrumordelaluz** [comment]
Hi @hali241997 I am closing this _issue_, but feel free to re-open or create a new one in any case.
